presentation/common/src/presentation-common/Update.ts

```diff
@@ -118,67 +119,97 @@ export type PartialHierarchyModification = NodeInsertionInfo | NodeDeletionInfo
 export namespace PartialHierarchyModification { // eslint-disable-line @typescript-eslint/no-redeclare
   /** Serialize given object to JSON. */
   export function toJSON(obj: PartialHierarchyModification): PartialHierarchyModificationJSON {
-    return {
-      ...obj,
-      node: Node.toJSON(obj.node),
-    };
+    switch (obj.type) {
+      case "Insert":
+        return {
+          type: "Insert",
+          parent: obj.parent === undefined ? undefined : NodeKey.toJSON(obj.parent),
+          position: obj.position,
+          node: Node.toJSON(obj.node),
+        };
+
+      case "Update":
+        return {
+          type: "Update",
+          target: NodeKey.toJSON(obj.target),
+          changes: Node.toPartialJSON(obj.changes),
+        };
+
+      case "Delete":
+        return {
+          type: "Delete",
+          target: NodeKey.toJSON(obj.target),
+        };
+    }
   }
 
   /** Deserialize given object from JSON */
   export function fromJSON(json: PartialHierarchyModificationJSON): PartialHierarchyModification {
-    return {
-      ...json,
-      node: Node.fromJSON(json.node),
-    };
+    switch (json.type) {
+      case "Insert":
+        return {
+          type: "Insert",
+          parent: json.parent === undefined ? undefined : NodeKey.fromJSON(json.parent),
+          position: json.position,
+          node: Node.fromJSON(json.node),
+        };
+
+      case "Update":
+        return {
+          type: "Update",
+          target: NodeKey.fromJSON(json.target),
+          changes: Node.fromPartialJSON(json.changes),
+        };
+
+      case "Delete":
+        return {
+          type: "Delete",
+          target: NodeKey.fromJSON(json.target),
+        };
+    }
   }
 }
 
 /** @alpha */
 export interface NodeInsertionInfo {
   type: "Insert";
+  parent?: NodeKey;
   position: number;
   node: Node;
 }
 
 /** @alpha */
 export interface NodeInsertionInfoJSON {
   type: "Insert";
+  parent?: NodeKey;

```

Comment:
Should be `NodeKeyJSON`

Reply:
Thanks, good catch. Fixed in 0e2568e0711d1fd7da221ba9920cba7fc0c64aa2.